invenio_rdm_records/services/config.py

```diff
@@ -50,13 +50,18 @@ class RDMRecordServiceConfig(RecordDraftServiceConfig):
             },
             'languages': {
                 'terms': {'field': 'metadata.languages.id'},
-            }
+            },
+            'record_status': {
+                'terms': {'field': 'metadata.status'},
+            },
         },
         post_filters={
             'subtype': terms_filter('metadata.resource_type.subtype'),
             'resource_type': terms_filter('metadata.resource_type.type'),
             'access_right': terms_filter('access.access_right'),
             'languages': terms_filter('metadata.languages.id'),
+            'record_status': terms_filter('metadata.status'),
+            'has_draft': terms_filter('metadata.has_draft'),
```

Comment:
I would not expose the has_draft in the REST API. This filter should be implicitly managed internally.